### ISSUE TITLE
Running large dataset with SqueezeMeta on a computing cluster

### ISSUE THREAD

**Issue by LunavdL** [jtamames/SqueezeMeta#482]
Hi,

I have a reasonably large dataset (94 samples, with a total of 4,347,241,600 reads). I am currently running it in the merged mode on our local server (RAM: 320 GiB (max: 384 GiB), CPU: 12 cores – 24 threads). Especially STEP1.5 -> 01.merge_assemblies.pl takes a very long time (it has been running for weeks and is still working on STEP1.5). 

I was wondering if I could speed up the process by running it on our computing cluster? Can SqueezeMeta run on multiple nodes simultaneously? 

Best regards,
Luna

**Comment by fpusan** [comment]
The seqmerge mode will be an improvement, but still will scale poorly when you have a lot of samples.
You can either try coassembly (likely to run out of memory), divide your data into smaller coassemblies (grouping the most similar samples into one coassembly), or just running the sequential mode.

**Comment by LunavdL** [comment]
Thank you! I tried the coassembly mode at first, but that indeed crashed.

I also started a subset of samples (from the same host species), but I am running that one on merged mode as well. It would be better to use coassembly when dividing the data into subsets? 
Can I merge all results in R if I process my data in subsets?   

I was also thinking I could remove duplication reads to reduce my file sizes (I am currently using my raw reads), but I am not sure if this is recommended with SqueezeMeta?

**Comment by fpusan** [comment]
Indeed, you can merge your results later with R. I would use co-assembly for the smaller subsets. I don't think removing the duplicates would help, the number of unique kmers in your dataset would remain the same so most likely memory usage by the assembler wouldn't decrease...

**Comment by LunavdL** [comment]
Thank you again! I have closed the issue.
